src/content/insertable-streams/webgpu/js/multi_video_worker.js

```diff
@@ -0,0 +1,248 @@
+        new Float32Array(vertexBuffer.getMappedRange()).set(rectVerts);
+        vertexBuffer.unmap();
+        context.configure({
+            device,
+            format: swapChainFormat
+        })
+
+        renderPipeline = device.createRenderPipeline({
+            vertex: {
+                module: device.createShaderModule({
+                    code: wgslShadersWorkers.vertex,
+                }),
+                entryPoint: 'main',
+                buffers: [
+                    {
+                        arrayStride: 20,
+                        attributes: [
+                            {
+                                // position
+                                shaderLocation: 0,
+                                offset: 0,
+                                format: 'float32x3',
+                            },
+                            {
+                                // uv
+                                shaderLocation: 1,
+                                offset: 12,
+                                format: 'float32x2',
+                            },
+                        ],
+                    },
+                ],
+            },
+            fragment: {
+                module: device.createShaderModule({
+                    code: wgslShadersWorkers.fragment,
+                }),
+                entryPoint: 'main',
+                targets: [
+                    {
+                        format: swapChainFormat,
+                    },
+                ],
+            },
+            primitive: {
+                topology: 'triangle-list',
+            },
+        });
+
+        videoTexture = device.createTexture({
+            size: {
+                width: 960,
+                height: 540,
+                depthOrArrayLayers: 1,
+            }, arrayLayerCount: 1,
+            mipLevelCount: 1,
+            sampleCount: 1,
+            dimension: '2d',
+            format: 'rgba8unorm',
+            usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.TEXTURE_BINDING |
+                GPUTextureUsage.RENDER_ATTACHMENT,
+        });
+
+        sampler = device.createSampler({
+            addressModeU: 'repeat',
+            addressModeV: 'repeat',
+            addressModeW: 'repeat',
+            magFilter: 'linear',
+            minFilter: 'linear',
+        });
+
+        postMessage({ result: 'Done' });
+    }
+    else if (operation === 'transform') {
+        const { videoStream, gumStream } = event.data;
+        const videoSource = videoStream.getReader();
+        const gumSource = gumStream.getReader();
+        if (videoSource === undefined || videoSource === null) {
+            console.log('[WebGPUWorker] videoSource is undefined or null.')
+            return;
+        }
+        if (gumSource === undefined || gumSource === null) {
+            console.log('[WebGPUWorker] gumSource is undefined or null.')
+            return;
+        }
+
+        while (true) {
```

Comment:
nit: need some way to terminate the loop and cancel videoSource/gumSource when we get the 'destroy' operation